### PR TITLE
[distroless] aws node-termination-handler to distroless

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

- [x] cloud-provider-aws/node-termination-handler
## Why do we need it, and what problem does it solve?

This PR is a part of moving Deckhouse to distroless.
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-aws
type: chore
summary: "`node-termination-handler` use distroless based image."
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

File: modules/030-cloud-provider-aws/images/node-termination-handler/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+shell:
+  beforeInstall:
+  - | 
+    apk upgrade --available --no-cache && \
+    apk add --no-cache ca-certificates git make 
+  install:
+  - export GO_VERSION=${GOLANG_VERSION}
+  - export GOPROXY={{ $.GOPROXY }}
+  - mkdir -p /src
+  - git clone --depth 1 --branch v1.5.0 https://github.com/aws/aws-node-termination-handler.git /src
```
Comment (RomanenkoDenys) [inline modules/030-cloud-provider-aws/images/node-termination-handler/werf.inc.yaml:26]: We use SOURCE_REPO here

File: candi/image_versions.yml [hunk 2]
```diff
@@ -29,3 +29,4 @@ BASE_SCRATCH: "spotify/scratch@sha256:3ab2f8af40415f6aec66bb953119a3bdc3ec4b542c
 BASE_SHELL_OPERATOR: "flant/shell-operator:v1.3.2@sha256:e837ce77f2943eb7b456d1f3059fa83625ef2b8602e597ed30218a91f3bb4842"
 BASE_UBUNTU_BIONIC: "ubuntu:bionic-20220531@sha256:2776b2dc16a72a7dabcbb0b0b70c181922c45e9a08aa740ddc65f53511863e32"
 BASE_UBUNTU: "ubuntu:jammy-20221130@sha256:c14c3b1242536729ce5227ff833144977b4e378723858fb73a4cf40ea6daaf6a"
+BASE_ALT: "alt:p10@sha256:f105773c682498700680d7cd61a702a4315c4235aee3622757591fd510fb8b4a"
```
Comment (RomanenkoDenys) [inline candi/image_versions.yml:32]: Why BASE_ALT is needed here ?

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some comments

**Comment by universal-itengineer** [comment]
Before sent termination-time
image

Sent termination-time
image

Node scheduled for disable
image

**Comment by universal-itengineer** [comment]
How to test:

1. Download  helm chart **amazon-ec2-metadata-mosk**  
`curl -L https://github.com/aws/amazon-ec2-metadata-mock/releases/download/v1.11.2/amazon-ec2-metadata-mock-1.11.2.tgz -o amazon-ec2-metadata-mock-1.11.2.tgz`
2. Install helm chart **amazon-ec2-metadata-mosk**  
`helm install amazon-ec2-metadata-mock amazon-ec2-metadata-mock-1.11.2.tgz --namespace d8-cloud-provider-aws`
3. Check that service exist
```
kubectl -n d8-cloud-provider-aws get svc -o wide
```
```
#Output
NAME                               TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE    SELECTOR
amazon-ec2-metadata-mock-service   ClusterIP   10.222.105.219   <none>        1338/TCP   106s   app.kubernetes.io/instance=amazon-ec2-metadata-mock
```
4. Configure daemonset `node-termination-handler` to get events from ~~aws~~ amazon-ec2-metadata-mock
```
kubectl -n d8-cloud-provider-aws edit daemonset node-termination-handler
```
```yaml
...
spec:
  containers:
    - name: aws-node-termination-handler
...
    env:
      - name: NODE_NAME
        ...
      - name: INSTANCE_METADATA_URL
        value: "http://10.222.105.219:1338" #example of meta instance url
...
```
5. Check service  
```
curl http://10.222.105.219:1338/latest/meta-data
```

```
#Output
ami-id
ami-launch-index
ami-manifest-path
block-device-mapping/
elastic-inference/
events/
hostname
iam/
instance-action
instance-id
instance-life-cycle
instance-type
kernel-id
local-hostname
local-ipv4
mac
network/
placement/
product-codes
public-hostname
public-ipv4
public-keys/
ramdisk-id
reservation-id
security-groups
services/
spot/
tags/
```
6. For example, send termination-time  
```
curl http://10.222.105.219:1338/latest/meta-data/spot/termination-time
```
```
#Output
2023-11-08T14:10:29Z
```
7. Check that node drain node  
```
kubectl get node
```
```
#Output
NAME                                            STATUS                     ROLES                  AGE   VERSION
distroless-pr6376-master-0                      Ready                      control-plane,master   84m   v1.25.15
distroless-pr6376-worker-dc42b8d2-676c4-8pc5c   Ready,SchedulingDisabled   worker                 76m   v1.25.15
```

**Review by name212 — APPROVED** [review]

**Pull request merged**